src/services/parser/bingParser.ts

```diff
@@ -0,0 +1,76 @@
+import { ISelectorId } from '@/types/common';
+
+import BaseParser from './baseParser';
+
+const bingSelectorIds: ISelectorId = {
+  adLinkElementAttribute: '.b_adurl',
+  adTopParent: '.b_ad, .b_adTop',
+  adWords: '.sb_ad, .sb_adTA',
+  linkElement: 'a',
+  adBottomParent: '.b_ad, .b_adBottom',
+  nonAdWords: '.b_algo',
+  linkElementAttribute: 'href',
+  nonAdParent: '#b_results',
+};
+
+class BingParser extends BaseParser {
+  selectorIds: ISelectorId;
+
+  constructor(htmlRawString: string) {
+    super(htmlRawString);
+    this.selectorIds = bingSelectorIds;
+  }
+
+  parseTopAdWordList() {
+    const urls: string[] = [];
+    const adElementsDom = this.htmlDocumentAPI(this.selectorIds.adTopParent);
+
+    const adUrlElements = this.htmlDocumentAPI(adElementsDom).find(
+      `${this.selectorIds.adTopParent} ${this.selectorIds.adLinkElementAttribute}`
+    );
+
+    adUrlElements.each((index, element) => {
```

Comment:
As we not use `index`

```suggestion
    adUrlElements.each((_, element) => {
```